test/jdk/tools/jmod/JmodTest.java

```diff
@@ -739,6 +739,79 @@ public void testTmpFileRemoved() throws IOException {
             });
     }
 
+    @Test
+    public void testCompressionLevel() throws IOException {
+        String cp = EXPLODED_DIR.resolve("foo").resolve("classes").toString();
+        Path jmod = MODS_DIR.resolve("foo.jmod");
+        FileUtils.deleteFileIfExistsWithRetry(jmod);
+
+        jmod("create",
+             "--class-path", cp,
+             "--compress", "zip-0",
+             jmod.toString())
+            .assertSuccess();
+
+        jmod("list",
+             "--compress", "zip-0",
+             jmod.toString())
+            .assertFailure()
+            .resultChecker(r -> {
+                assertTrue(r.output.contains("--compress is only accepted with create mode"), "Error message printed");
+            });
+
+        FileUtils.deleteFileIfExistsWithRetry(jmod);
+
+        jmod("create",
+             "--class-path", cp,
+             "--compress", "zip-9",
+             jmod.toString())
+            .assertSuccess();
+
+        FileUtils.deleteFileIfExistsWithRetry(jmod);
+
+        jmod("create",
+             "--class-path", cp,
+             "--compress", "zip--1",
+             jmod.toString())
+            .assertFailure()
+            .resultChecker(r -> {
+                assertTrue(r.output.contains("--compress value is invalid"), "Error message printed");
+            });
+
+        FileUtils.deleteFileIfExistsWithRetry(jmod);
+
+        jmod("create",
+             "--class-path", cp,
+             "--compress", "zip-1-something",
+             jmod.toString())
+            .assertFailure()
+            .resultChecker(r -> {
+                assertTrue(r.output.contains("--compress value is invalid"), "Error message printed");
+            });
+
+        FileUtils.deleteFileIfExistsWithRetry(jmod);
+
+        jmod("create",
+             "--class-path", cp,
+             "--compress", "zip-10",
+             jmod.toString())
+            .assertFailure()
+            .resultChecker(r -> {
+                assertTrue(r.output.contains("--compress value is invalid"), "Error message printed");
+            });
+
+        FileUtils.deleteFileIfExistsWithRetry(jmod);
+
+        jmod("create",
+             "--class-path", cp,
+             "--compress", "test",
```

Comment:
good to have a test case verifying "test-0" value.

Reply:
Yes, added.